src/Microsoft.Restier.Core/Query/QueryModelReference.cs

```diff
@@ -240,10 +280,19 @@ public IEdmProperty Property
         {
             get
             {
-                var structuredType = this.Source.Type as IEdmStructuredType;
-                if (structuredType != null)
+                if (property != null)
+                {
+                    return property;
+                }
+
+                if (Source != null)
```

Comment:
please investigate more on this part of logic...there is an `Ensure.NotNull(property)` on line 216 so the following logic will never get executed.